lfs.c

```diff
@@ -2584,6 +2585,12 @@ static int lfs_file_rawopencfg(lfs_t *lfs, lfs_file_t *file,
 
     // zero to avoid information leak
     lfs_cache_zero(lfs, &file->cache);
+#else
+   // local buffer to be shared among all calls
+   static uint8_t no_malloc_buffer[1024];
```

Comment:
I believe that the buffer should have a size of `lfs->cfg->cache_size` bytes, which is probably much less than 1024.

Also, as far as I can tell, this would mean that all files (and worse, all file systems) use the same buffer, which is inherently unsafe in multi-tasking (or multi-threaded) environments.

Reply:
sorry, you're right
I've pushed this commit in a rush which is bad
you should through it away
I'll provide a better one